opus/opus.go

```diff
@@ -178,6 +180,13 @@ func (e *encoder) WriteSample(in media.PCM16Sample) error {
 	if err != nil {
 		return err
 	}
+
+	// opus_encode() returns the number of bytes actually written to the packet.
+	// The return value can be negative, which indicates that an error has occurred.
+	// If the return value is 1 byte, then the packet does not need to be transmitted (DTX).
+	if n == 1 && e.useDtx {
+		return nil
+	}
```

Comment:
:TIL: I thought 1 byte should be transmitted to signal background noise level. I think I have seen 1 byte packets comes in. But, `libwebrtc` mentions this also (https://source.chromium.org/chromium/chromium/src/+/main:media/audio/audio_opus_encoder.cc;drc=dfb49a410ff213451f72db8d792973550025033e;l=288). Where are the 1 byte packets coming from into the server? Have to check again if I am not remembering something properly.

Reply:
JS SDK sample app does DTX. I think I checked with that. Will test out and check packet sizes with DTX enabled.

Reply:
DTX packets (or the packet before DTX kicks in) is 1 byte when trying with JS SDK sample app (setting `red: false` in the demo app) and logging on SFU side. Puzzled what opus_encode returns and how that is getting sent. Definitely does not jive with opus_encode() doc or libwebrtc code.